lib/compiler/passes/generate-bytecode.js

```diff
@@ -618,7 +618,7 @@ function generateBytecode(ast, options) {
         [op.SILENT_FAILS_ON],
         generate(node.expression, context),
         [op.SILENT_FAILS_OFF],
-        buildCondition(match, [op.IF_ERROR], [op.FAIL, nameIndex], [])
+        buildCondition(-match, [op.IF_ERROR], [op.FAIL, nameIndex], [])
       );
```

Comment:
Good catch! The bug probably was introduced during rebase while adapt initial PR to the new code after another merged PR.

Actually, the whole `buildSequence` could be replaced by `[op.FAIL, nameIndex]` if node always fail and there are no actions or semantic predicates (because they can have side effects), but anyway, is is better to do in a separate pass.

Reply:
> if node always fail and there are no actions or semantic predicates (because they can have side effects)

There are already [comments](https://github.com/peggyjs/peggy/blob/63b913da87cff16ba9d2275b2f48d1dba58b2b33/lib/compiler/passes/generate-bytecode.js#L640-L643) about adding side-effect analysis so we can do that kind of thing...